### PR TITLE
Rename CmdDef to QasmToSchedDef

### LINKED ISSUE #2681: Refactor: `CmdDef` -> `QasmToSchedDef`
### What is the expected enhancement?

Part of the #2678 Epic. Repeated below:

Change: The name of `CmdDef` class
Proposal: `QasmToSchedDef`
Rationale: The class currently describes all QuantumCircuit Instruction implementations for certain qubits for a given backend as `Schedule`s. A `Schedule` does contain `Instruction`s which contain `Command`s, but `Command`s are not fundamentally what we are gaining from this class, since `Schedule`s each contain multiple `Command`s, are particular for `Channel`s, and have specific timing. The new name avoids conflating `Schedule`s and `Command`s.
`CmdDef` is currently the name due to the fixed interface at the transport layer. With Issue #2680, we can decouple this constraint.

### PR DESCRIPTION
**Pull request by lcapelluto**
### Summary

Closes #2681. `cmd_def.py` will have to be deleted later if we deprecate like this.

### Details and comments
Change `CmdDef` -> `QasmToSchedDef` because this object is responsible for translating QASM operations (gates and measures) to Schedules, as defined by the provided backend.

Related changes (beyond comments):
`CmdDef.cmds` is replaced by `QasmToSchedDef.gates`
`CmdDef.cmd_qubits` is replaced by `QasmToSchedDef.gate_qubits`
Private attribute `CmdDef._cmd_dict` replaced by `CmdDef._gate_dict`

I am not set on these ^^^ names. Except for measures, we are always working with Gate -> Schedule here. It's sometimes awk to say "QASM operation, gates and measures" all the time. I think "Gate" is still a pretty clear and concise way to say "Circuit Element". Will it be too confusing that Measures are not gates? Is there a better name for this? Let's get it right -- I welcome suggestions.

Not changed:
`cmd_def` is still used in the context of `backend.defaults().cmd_def` and `backend.defaults().build_cmd_def()`

### CODE REVIEW

File: qiskit/pulse/qasm_def.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
+# -*- coding: utf-8 -*-
```
Comment (taalexander) [inline qiskit/pulse/qasm_def.py]: I think this name should reflect closer to the class name since it is a module for only a single class.

Comment (lcapelluto) [reply qiskit/pulse/qasm_def.py]: yes you are right 👍

**Comment by lcapelluto** [comment]
@taalexander 
> I think the measure is a good counter-example to why this should not be named gates. To me within the context of quantum computing Gate implies a unitary action which the measurement violates. There will also be more counterexamples as we continue to add more classical control flow.

I agree here. As an aside I hope that classical control operations don't end up in the CmdDef somehow. Yeah, no perfect name for this. Maybe we can sit on it a few days and brainstorm.

**Comment by dcmckayibm** [comment]
What about just `PulseMapper`?

**Comment by taalexander** [comment]
I can get down with `PulseMapper`. Its ambiguous enough that it shouldn't be confused with something else.

**Comment by lcapelluto** [comment]
Added `on hold`, because this shouldn't be merged unless we all feel good about all the names (being that that is the only reason for this PR). I'll revisit this next week.

**Comment by lcapelluto** [comment]
I think `CmdDef` will be wrapped in a higher level interface. I don't think we have to change the name.

**Comment by taalexander** [comment]
Could you please elaborate on the above?

**Pull request closed without merging**